lib/network-layer/src/Cardano/Wallet/Network/Implementation.hs

```diff
@@ -659,6 +661,10 @@ withNodeNetworkLayerBase
                     return res
                 Nothing -> pure $ StakePoolsSummary 0 mempty mempty
 
+        _getUTxOByTxIn queue ins =
```

Comment:
I agree that it's worth adding the special case — we don't want to do an IO roundtrip over the node socket if we already know the answer.

But which era should the result be in?

At the very least, calling `getUTxOByTxIn` will entail a query about the current node era — which we would need to cache in order to avoid sending something to the node.

Reply:
> But which era should the result be in?

Right, good point. Luckily we already have access to a cached value of the era, so it should be easy.

I think we could also imagine using your `Data.These` here with something isomorophic to `Maybe (Data.These (UTxO ConwayEra) (UTxO BabbageEra))`, to be able to return `Just $ These mempty mempty`, but doesn't seem worth it for this problem alone (particularly as `Cardano.Wallet.balanceTx` is still uncertain about the `UTxO`'s era even when the node was queried)

Reply:
[1272251](https://github.com/cardano-foundation/cardano-wallet/pull/4630/commits/1272251d600237c4b00071ae0e2e4d6ef3f9410e)